doc/life-cycles/digest.dot

```diff
@@ -6,28 +6,30 @@ digraph digest {
     initialised [label=initialised, fontcolor="#c94c4c"];
     updated [label=updated, fontcolor="#c94c4c"];
     finaled [label="finaled", fontcolor="#c94c4c"];
+    squeezed [label="squeezed", fontcolor="#c94c4c"];
     end [label="freed", color="#deeaee", style="filled"];
 
     begin -> newed [label="EVP_MD_CTX_new"];
-    newed -> initialised [label="EVP_DigestInit"];
-    initialised -> updated [label="EVP_DigestUpdate", weight=3];
+    newed -> initialised [label="EVP_DigestInit", weight=100];
+    initialised -> updated [label="EVP_DigestUpdate", weight=100];
```

Comment:
Is the transition `initialised -> finaled` missing?

Reply:
Yep, they should be added. Although not useful much they are legitimate.

Reply:
Some of the state checkings are also missing in the generic implementations in the sha3 provider. I'll add a proposal for it tomorrow to my branch,

Reply:
I'd avoid adding `initialised -> squeezed` since it's not useful.

I don't think we can avoid allowing `initialised -> finaled` but we don't have to advertise it.

Reply:
> I'd avoid adding `initialised -> squeezed` since it's not useful.

The current implementation for s390x accepts squeeze calls from XOF_STATE_INIT. Should I change that?

Reply:
I don't think it needs changing.

Reply:
Ok, fine with me.